src/server/server.rs

```diff
@@ -391,28 +387,69 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver> Server<T, S> {
             return self.write_data(event_loop, token, data, None);
         }
 
-        // No connection, resolve the store address and then send it.
-        self.resolve_store(store_id, data);
+        // No connection, append to pending list, resolve the store address,
+        // and then send it.
+        let need_resolve;
```

Comment:
maybe yes, but no need.
If we drop it, raft may not know how to do but retry after timeout. Or it is not suitable to return unreachable to raft too here because we may resolve ok and can send to remote.